network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt

```diff
@@ -37,67 +38,70 @@ import javax.net.ssl.SSLContext
-                hosts.forEach { host ->
-                    add(host, cert)
+        // OkHttp doesn't support configuring ping intervals dynamically,
+        // so they must be set when creating the Engine
+        // See https://youtrack.jetbrains.com/issue/KTOR-4752
+        if (isProxyRequired(serverConfigDTOApiProxy)) {
+            if (serverConfigDTOApiProxy?.needsAuthentication == true) {
+                if (proxyCredentials == null) error("Credentials does not exist")
+                with(proxyCredentials) {
+                    Authenticator.setDefault(object : Authenticator() {
+                        override fun getPasswordAuthentication(): PasswordAuthentication {
+                            return PasswordAuthentication(username, password?.toCharArray())
+                        }
+                    })
                 }
             }
-        }.build()
-        okHttpClient.certificatePinner(certPinner)
-    }
-
-    if (ignoreSSLCertificates) okHttpClient.ignoreAllSSLErrors()
 
-    okHttpClient.pingInterval(WEBSOCKET_PING_INTERVAL_MILLIS, TimeUnit.MILLISECONDS)
-        .connectTimeout(WEBSOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
-        .readTimeout(WEBSOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
-        .writeTimeout(WEBSOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
+            val proxy = Proxy(
+                Proxy.Type.SOCKS,
+                InetSocketAddress.createUnresolved(serverConfigDTOApiProxy?.host, serverConfigDTOApiProxy!!.port)
+            )
 
-    // OkHttp doesn't support configuring ping intervals dynamically,
-    // so they must be set when creating the Engine
-    // See https://youtrack.jetbrains.com/issue/KTOR-4752
-    if (isProxyRequired(serverConfigDTOApiProxy)) {
-        if (serverConfigDTOApiProxy?.needsAuthentication == true) {
-            if (proxyCredentials == null) error("Credentials does not exist")
-            with(proxyCredentials) {
-                Authenticator.setDefault(object : Authenticator() {
-                    override fun getPasswordAuthentication(): PasswordAuthentication {
-                        return PasswordAuthentication(username, password?.toCharArray())
-                    }
-                })
-            }
+            proxy(proxy)
         }
 
-        val proxy = Proxy(
-            Proxy.Type.SOCKS,
-            InetSocketAddress.createUnresolved(serverConfigDTOApiProxy?.host, serverConfigDTOApiProxy!!.port)
-        )
-
-        val clientWithProxy = okHttpClient.proxy(proxy).build()
-        preconfigured = clientWithProxy
-        webSocketFactory = KaliumWebSocketFactory(clientWithProxy)
-
-    } else {
-        val client = okHttpClient.build()
-        preconfigured = client
-        webSocketFactory = KaliumWebSocketFactory(client)
+    }.also {
+        preconfigured = it
+        webSocketFactory = KaliumWebSocketFactory(it)
     }
 }
 
-fun OkHttpClient.Builder.ignoreAllSSLErrors(): OkHttpClient.Builder {
```

Comment:
this function was creating a new builder while it was not used

Reply:
good catch !